pallets/msa/src/lib.rs

```diff
@@ -1209,6 +1317,391 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Propose to be a provider.  Creates a proposal for council approval to create a provider from a MSA
+		///
+		/// # Errors
+		/// - [`NoKeyExists`](Error::NoKeyExists) - If there is not MSA for `origin`.
+		#[pallet::call_index(19)]
+		#[pallet::weight(T::WeightInfo::propose_to_be_provider_v2())]
+		pub fn propose_to_be_provider_v2(
+			origin: OriginFor<T>,
+			payload: ProviderRegistryEntry<
+				T::MaxProviderNameSize,
+				T::MaxLanguageCodeSize,
+				T::MaxLogoCidSize,
+				T::MaxLocaleCount,
+			>,
+		) -> DispatchResult {
+			let proposer = ensure_signed(origin)?;
+			Self::ensure_valid_msa_key(&proposer)?;
+			Self::ensure_correct_cids(&payload)?;
```

Comment:
I think maybe we can skip validating the CIDs here. It gets done in the wrapped extrinsic; why pay for the compute cost of the same calculations twice?